### PR TITLE
Fix backend target in docker compose

### PR DESCRIPTION
**Pull request by salvacorts**
**What this PR does / why we need it**:
At https://github.com/grafana/loki/pull/9899 added the new `backend` SSD target to the docker compose we ship at `production/docker`. Later on, https://github.com/grafana/loki/pull/8836 removed it since the current stable version of Loki was 2.7.3 and `backend` was introduced with 2.8 (RC at the time of this PR).

The backend target is now GA, this PR enables it back in our provided docker compose.

Additionally, the following improvements/updates are made:
- Move storage config from `common.storage.s3` to `storage.aws`
- Remove schema period using boltdb-shipper
- Update tsdb schema to latest (v13)
- Expose minio web console

### CODE REVIEW

File: production/docker/config/loki.yaml [hunk 1]
```diff
@@ -71,17 +76,10 @@ ruler:
 
 schema_config:
   configs:
-  - from: 2020-08-01
```
[the remark this answers is not in the document]
Comment (salvacorts) [reply production/docker/config/loki.yaml:74]: Good call. Added that back.

Still I wonder if we can expect users to update this config for their use case. Two reasons:
- Keeping backwards compatibility here adds maintainability overhead.
- I don't think this compose file is production ready as it is. E.g. it uses minio with weak credentials, it uses a fake log generator and example rules/alerts.

I feel this is more like an example than a production ready bundle.

**Review by trevorwhitney — APPROVED** [review]
ssd stuff looks good to me, no other comments other than what Danny has already pointed out. thanks for fixing this up!

File: production/docker/config/loki.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (dannykopping) [reply production/docker/config/loki.yaml]: Yeah I'm unsure too, but I think when in doubt it's best to maintain the status quo.

**Review by dannykopping — APPROVED** [review]
LGTM!

**Pull request merged**